### ISSUE TITLE
Support querying by aria-labelledby for tags that aren't <label>

### ISSUE THREAD

**Issue by alexkrolick** [kentcdodds/dom-testing-library#120]
### Describe the feature you'd like:

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute

The `aria-labelledby` attribute doesn't require the labeling element to be a `<label>` tag, but dom-testing-lib only searches for `label`s in `queryByLabelText`.

I'd like to add a way to query for anything by ARIA label, which would allow doing things like selecting a card item by its heading text and then doing scoped queries for related elements like an "edit" button.

This would largely replace the need for workarounds like `getByText` -> `.$closest()` parent testid.

**Comment by kentcdodds** [comment]
Hey Alex 👋 

Interesting. So would making it so you can pass a custom selector to [this call](https://github.com/kentcdodds/dom-testing-library/blob/7b57f7313c3e751ad10e7ac8198ccc9b381cf82f/src/queries.js#L21) be sufficient?

**Comment by alexkrolick** [comment]
I don't think so, I think it should be easier to do to encourage the use of `aria-labelledby` instead of testid.

**Comment by alexkrolick** [comment]
Plus for performance I think it makes sense to work backwards from nodes with the `aria-labelledby` attribute instead of traversing all nodes first

**Comment by alexkrolick** [comment]
Would it be a breaking change to change `getByLabelText` to this:

- Remove this part: https://github.com/kentcdodds/dom-testing-library/blob/7b57f7313c3e751ad10e7ac8198ccc9b381cf82f/src/queries.js#L47-L52

- Add something that does the same thing using `queryAllByText` as the initial filter?

```js
const possibleLabelNodes = queryAllByText(query);

const ariaLabelledNodes = possibleLabelNodes.reduce(
  (allLabeledNodes, nextLabelNode) => {
    const labelId = nextLabelNode.getAttribute("id");
    if (labelId === undefined) return allLabeledNodes;

    // ARIA labels can label multiple elements
    const labelledNodes = container.querySelectorAll(
      `[aria-labelledby~="${labelId}"]`,
    );

    return allLabeledNodes.concat(labelledNodes);
  },
);
```

- Combine the results

```js
return _.uniq(labelNodes, ariaLabelledNodes)
```

**Comment by kentcdodds** [comment]
That seems reasonable 👍
